### PR TITLE
Mobile product listing tweaks

### LINKED ISSUE #6080: [mobile ux] Remaining tweaks for product listings epic
## Description
Work on the shop product listings has been happening across a number of different story cards connected to the epic #4597 and also [this single testing document](https://docs.google.com/document/d/1WZZSuMCLHiZ_UY_nLJZ8MI-HGadTQmVty7zXeS7H6VE/edit?usp=sharing).

The below list of work to be done is all that remains to complete the epic, and therefore merge the product list branch to master and release the changes. The detail of these changes can be found in the testing doc above.

*Note: the product listing branch cannot be in a release prior to the image resizing server-side work that @mkllnk is managing alongside of the actual styling changes (ie those things below).

### Price breakdown modal

- [x] Price breakdown modal is missing it's title (see testing doc for screenshots)

- [x] Padding/margins in the modal aren't quite right (see testing docs for css changes provided by @yukoosawa)

- [x] Price breakdown modal extends to the far right side of the screen on mobile, requires right side padding (see testing doc for screenshots)

### Adding bulk products to cart

- [x] The bulk visual indicator that displays over the product image is missing (see testing doc for screenshots)

- [x] Add bulk products modal is too wide on desktop and tablet (see testing doc for screenshots)

- [x] Disabled button colour needs to be light grey (#bbbbbb) rather than faded orange.

### Displaying product property descriptions when hover/click on a property tag

- [x] Property description hover box caret does not display on mobile/tablet (see testing doc for screenshots)

- [ ] Clicking on the property tag on mobile when the description is open closes the description (ie. first click on tag opens it, second closes it)

### Variants

- [x] When a variant name wraps on multiple lines the line height is too big (see testing doc for screenshots)
- [x] Unit size: update the text to: `font-style: italic` to help separate it from the variant name

### Product list layout and styling

- [ ] Product pricing detail on listings is not in correct position on tablet (see testing doc for screenshots)

- [x] The "from" before the producer name on the listing should be grey, not black.

Design: https://app.zeplin.io/project/59532764a6c78ff3c534b372/dashboard

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

Closes #6080

A few more tweaks to the new product listing design for mobile. They were found in the last testing round and documented in:
https://docs.google.com/document/d/1WZZSuMCLHiZ_UY_nLJZ8MI-HGadTQmVty7zXeS7H6VE

Two issue were skipped because the effort to implement them seems out of proportion to the problem.

#### What should we test?

The remaining issues

##### Price breakdown modal
- [x] Price breakdown modal is missing it's title (see testing doc for screenshots)
- [x] Padding/margins in the modal aren't quite right (see testing docs for css changes provided by @yukoosawa)
- [x] Price breakdown modal extends to the far right side of the screen on mobile, requires right side padding (see testing doc for screenshots)

##### Adding bulk products to cart

- [x] The bulk visual indicator that displays over the product image is missing (see testing doc for screenshots)
- [x] Add bulk products modal is too wide on desktop and tablet (see testing doc for screenshots)
- [x] Disabled button colour needs to be light grey (#bbbbbb) rather than faded orange.

##### Displaying product property descriptions when hover/click on a property tag

- [x] Property description hover box caret does not display on mobile/tablet (see testing doc for screenshots)

##### Variants

- [x] When a variant name wraps on multiple lines the line height is too big (see testing doc for screenshots)
- [x] Unit size: update the text to: `font-style: italic` to help separate it from the variant name

##### Product list layout and styling

- [x] The "from" before the producer name on the listing should be grey, not black.

Reference design: https://app.zeplin.io/project/59532764a6c78ff3c534b372/dashboard

### CODE REVIEW

File: app/assets/javascripts/templates/price_breakdown.html.haml [hunk 1]
```diff
@@ -2,6 +2,7 @@
   %span.joyride-nub.top
   .background{ng: {click: "tt_isOpen = false"}}
   .joyride-content-wrapper
+    %h6= t("price_breakdown")
```
Comment (luisramos0) [inline app/assets/javascripts/templates/price_breakdown.html.haml:5]: Ideally we would use a specific namespace here under js. so that we continue our slow move into https://github.com/openfoodfoundation/openfoodnetwork/issues/2721

File: app/assets/stylesheets/darkswarm/_shop-popovers.scss [hunk 2]
```diff
@@ -4,23 +4,27 @@
...
+    padding: .625rem;
     width: 98%;
     background-color: $grey-800;
     color: white;
   }
 
-  h1, h2, h3, h4, h5, h6 {
-    color: #1f1f1f;
+  h6 {
+    font-size: 0.937rem;
```
Comment (luisramos0) [inline app/assets/stylesheets/darkswarm/_shop-popovers.scss:23]: this is 1/16th under the unit :+1:

File: app/assets/stylesheets/darkswarm/_shop-popovers.scss [hunk 3]
```diff
@@ -45,14 +49,15 @@
   }
 
   li {
+    line-height: 1;
     border-bottom: .1em solid $grey-700;
-    padding: 0 0.25rem;
+    padding: 0.625rem 0;
```
Comment (luisramos0) [inline app/assets/stylesheets/darkswarm/_shop-popovers.scss:54]: should we always us .625 or 0.625. I am not sure codeclimate says something about this but we should have a standard :+1:

File: app/assets/stylesheets/darkswarm/_shop-product-rows.scss [hunk 4]
```diff
@@ -105,6 +107,7 @@
           }
 
           .product-producer {
+            color: #888;
```
Comment (luisramos0) [inline app/assets/stylesheets/darkswarm/_shop-product-rows.scss:110]: I think colors should always be named variables.
In this case I think this can go to
app/assets/stylesheets/darkswarm/branding.scss
and be called $grey-550: #888;

File: app/views/shop/products/_summary.html.haml [hunk 5]
```diff
@@ -1,5 +1,7 @@
 .product-thumb
   %a{"ng-click" => "triggerProductModal()"}
+    %span.product-thumb__bulk-label{"ng-if" => "::product.group_buy"}
+      = t("products_bulk")
```
Comment (luisramos0) [inline app/views/shop/products/_summary.html.haml:4]: Ideally this would be a relative path in the translations file.

**Review by luisramos0 — APPROVED** [review]
Nice, almost done!!!
I left a few minor comments.

File: app/assets/stylesheets/darkswarm/_shop-popovers.scss (hunk 3 above)
Comment (mkllnk) [reply app/assets/stylesheets/darkswarm/_shop-popovers.scss:54]: Good point. We probably deactivated Code Climate for this because I remember that it preferred `.625`. I personally don't have a preference.

File: app/assets/stylesheets/darkswarm/_shop-product-rows.scss (hunk 4 above)
Comment (mkllnk) [reply app/assets/stylesheets/darkswarm/_shop-product-rows.scss:110]: I was wondering about that. I will add it. :-)

File: app/views/shop/products/_summary.html.haml (hunk 5 above)
Comment (mkllnk) [reply app/views/shop/products/_summary.html.haml:4]: Totally. I must have thought that I was in a JS template. I fixed this in a newer commit.

File: app/assets/javascripts/templates/price_breakdown.html.haml (hunk 1 above)
Comment (mkllnk) [reply app/assets/javascripts/templates/price_breakdown.html.haml:5]: Yes. More importantly, I realised that this is a JS template and we need to translate in JS to show other languages than the instance's default. I fixed that in another commit.

File: app/assets/stylesheets/darkswarm/_shop-popovers.scss (hunk 3 above)
Comment (luisramos0) [reply app/assets/stylesheets/darkswarm/_shop-popovers.scss:54]: ok, lets use .625 :+1:

**Review by luisramos0 — APPROVED** [review]
:ok_hand:

**Review by sauloperez — APPROVED** [review]
I'm approving because there's no point in holding it all because of that small comment. I'd like to address it later on though.

**Comment by luisramos0** [comment]
Note: the target branch here is not master but the product list branch. I think in some cases we have merged to that branch and then tested there. I assume @mkllnk will know :+1:

**Comment by filipefurtad0** [comment]
Hi @mkllnk,

I tested the remaining issues you mention. Used Browserstack for testing on a mobile device - tested on Iphone 6S - 11 Safari and desktop Ubuntu 20.04 Firefox 81.

1) **Price breakdown modal**
a) Price breakdown modal is missing it's title (see testing doc for screenshots)
b) Padding/margins in the modal aren't quite right (see testing docs for css changes provided by @yukoosawa)
c) Price breakdown modal extends to the far right side of the screen on mobile, requires right side padding (see testing doc for screenshots)

Issue a) b) and c) seem to be addressed - see pic below: 

image
 
It would be useful to open a console on Safari/Chrome mobile to check the padding values (issue 1.b). I googled around a bit, but I can't seem to make this happen on Browserstack. Any ideas?



2) Adding bulk products to cart

a) The bulk visual indicator that displays over the product image is missing (see testing doc for screenshots)
b) Add bulk products modal is too wide on desktop and tablet (see testing doc for screenshots)
c) Disabled button colour needs to be light grey (#bbbbbb) rather than faded orange.


Issue a) Bulk indicator is now over the product image
image


Issue b) Bulk products modal now is less broad - looks great! 
image

Issue c) I am not sure this issue is correctly addressed. From the pics, only the two right side buttons should be disabled, right? It now look like this:

Desktop
image

Mobile
image




3) Displaying product property descriptions when hover/click on a property tag

a) Property description hover box caret does not display on mobile/tablet (see testing doc for screenshots)

This now seems displayed on mobile:
image

But I am unsure of this point: I Yuko's notes, this seems already to be the case, so I see no change here.

4) Variants

a) When a variant name wraps on multiple lines the line height is too big (see testing doc for screenshots)
b) Unit size: update the text to: font-style: italic to help separate it from the variant name

a) and b) are addressed, see pic below:

image

5) Product list layout and styling

a) The "from" before the producer name on the listing should be grey, not black.

Looks good:
image


In summary - this looks awesome! :tada: 

I'd need feedback on
2.c) should the PR be merged, and an issue opened for this one?
3.a) I'm not sure to see the correct change (or any change) to be introduced  

Issues found:
I noticed the OFN logo floats around, this is not the case on other staging servers, but is visible as well in Yuko's notes, for example here:

image

Is this OK? I didn't find an issue for this - should I open one? I observe this as well:
image

**Comment by sigmundpetersen** [comment]
I staged this on fr-staging and the floating logo issue looks OK there.
So I think that last issue found is staging specific to the logo on au-staging (i.e. not an issue with the PR).

**Comment by filipefurtad0** [comment]
Hi @sigmundpetersen , thanks for checking this :+1: Yes, indeed it looks fine on staging-FR.

But should 1:1 aspect ratio images be displayed in this way, in this section? Throughout the app, images are resized (ex. products) or cropped to a given format (ex. the promo image) and adjusted to a specific aspect ratio. In the example above, it seems the image is being resized in width (=75), but not in height. So, I wonder if (and how) this could be improved in this case, by resizing or cropping square images...

**Comment by Matt-Yorkley** [comment]
> it seems the image is being resized in width (=75), but not in height.

This is actually specific to au-staging, I think the logo file is just outdated and doesn't match the file used on other instances. You can upload a new one :+1: I don't think it's related to this PR at all

**Comment by Matt-Yorkley** [comment]
> a) Property description hover box caret does not display on mobile/tablet (see testing doc for screenshots)

"Hover" isn't really a thing on mobile...

**Comment by filipefurtad0** [comment]
Yes, I agree this is not introduced by the PR and choosing an other pic would solve this :+1: 

Also, and on point 3.a) yeah, I think this would not prevent merging. I've spotted no change here but I guess if any changed is still required it can be added afterwards.  

It's rather point 2.c) which I'm unsure of, since this PR makes the buttons look slightly worse, both on mobile and desktop. Maybe a decision for @openfoodfoundation/train-drivers-product-owners ?

**Comment by Matt-Yorkley** [comment]
I'll fix it now...

**Comment by filipefurtad0** [comment]
Thanks @Matt-Yorkley - moving to In Dev.

**Comment by Matt-Yorkley** [comment]
Fixed. It's a tiny tweak, I think it can go back to `test-ready` @filipefurtad0 :+1:

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,
I had look after your commit - point it looks good now:

- The color and the behavior seem correct. If "Min quantity" is set to zero, then only the plus button is active:

image

- Adding items changes the color of the inactive buttons:

image

Everything else looks as previously documented.

Thanks again @sigmundpetersen and @Matt-Yorkley for your feedback on this PR.
Moving to ready to go.

**Comment by luisramos0** [comment]
Nice! @Matt-Yorkley is this supposed to be merged into mobile-5-product-listings now?

**Comment by Matt-Yorkley** [comment]
I think so, then we just need a PR for merging the branch into master (which might involve some conflicts etc).

**Comment by luisramos0** [comment]
ok,  @mkllnk can you take care of that?

**Comment by daniellemoorhead** [comment]
Hey all, remember that te image resizing work has to be done on servers before the product listing branch can go into a release. I know Maikel knows that but making sure it’s clear for everyone :)

**Comment by mkllnk** [comment]
Awesome work, people! :tada: I'm going to merge this into the feature branch and I will open a new pull request for master. It will be blocked by the image standardisation work though.
